tests/components/notify/test_notify.py

```diff
@@ -1,145 +1,557 @@
-    assert notify.DOMAIN not in hass.config.components
-    mock_demo_notify.return_value = None
-    await discovery.async_load_platform(
-        hass, "notify", "demo", {"test_key": "test_val"}, {"notify": {}}
+    await async_load_platform(
+        hass,
+        "notify",
+        "testnotifyinvalid",
+        {"notify": [{"platform": "testnotifyinvalid"}]},
+        hass_config={"notify": [{"platform": "testnotifyinvalid"}]},
     )
     await hass.async_block_till_done()
-    assert notify.DOMAIN in hass.config.components
-    assert mock_demo_notify.called
-    assert mock_demo_notify.mock_calls[0][1] == (
+    assert "Unknown notification service specified" in caplog.text
+
+
+async def test_platform_setup_with_error(
+    hass: HomeAssistant, caplog: pytest.LogCaptureFixture, tmp_path: Path
+) -> None:
+    """Test service setup with an invalid setup."""
+
+    async def async_get_service(hass, config, discovery_info=None):
+        """Return None for an invalid notify service."""
+        raise Exception("Setup error")
+
+    mock_notify_platform(
+        hass, tmp_path, "testnotify", async_get_service=async_get_service
+    )
+    # Setup the second testnotify2 platform dynamically
+    await async_load_platform(
         hass,
+        "notify",
+        "testnotify",
         {},
-        {"test_key": "test_val"},
+        hass_config={"notify": [{"platform": "testnotify"}]},
+    )
+    await hass.async_block_till_done()
+    assert "Error setting up platform testnotify" in caplog.text
+
+
+async def test_reload_with_notify_builtin_platform_reload(
+    hass: HomeAssistant, caplog: pytest.LogCaptureFixture, tmp_path: Path
+) -> None:
+    """Test reload using the legacy notify platform reload method."""
+
+    async def async_get_service(hass, config, discovery_info=None):
+        """Get notify service for mocked platform."""
+        targetlist = {"a": 1, "b": 2}
+        return NotificationService(hass, targetlist, "testnotify")
+
+    # platform with service
+    mock_notify_platform(
+        hass, tmp_path, "testnotify", async_get_service=async_get_service
     )
 
+    # Perform a reload using the notify module for testnotify (without services)
+    await notify.async_reload(hass, "testnotify")
 
-async def test_sending_none_message(hass: HomeAssistant, events) -> None:
+    # Setup the platform
+    await async_setup_component(
+        hass, "notify", {"notify": [{"platform": "testnotify"}]}
+    )
+    await hass.async_block_till_done()
+    assert hass.services.has_service(notify.DOMAIN, "testnotify_a")
+    assert hass.services.has_service(notify.DOMAIN, "testnotify_b")
+
+    # Perform a reload using the notify module for testnotify (with services)
+    await notify.async_reload(hass, "testnotify")
+    assert hass.services.has_service(notify.DOMAIN, "testnotify_a")
+    assert hass.services.has_service(notify.DOMAIN, "testnotify_b")
+
+
+async def test_setup_platform_and_reload(
```

Comment:
Why are there new tests in this PR? This PR should just move the legacy tests to a new module and make them pass. No new additions should be made. Please do those separately.

Reply:
There are no new tests, the extra tests were moved from `test_init.py` as they are for the legacy service.
The tests for the demo platform were added, but needed refactoring.

Reply:
It's not possible to review this PR in an easy way either as a whole or the individual commits.

I suggest you start over in a fresh branch. One commit to move the tests and then one commit to make them pass. Don't remove any tests or add any tests or change name of any tests.

Reply:
Sorry for that, I started splitting the tests on the notify platform, but I should have started moving the legacy demo tests first.

Reply:
This PR only processes the tests from the demo integration that are kept.
It moves, updates the tests.
Then it restores the original demo test file back for coverage.